### ISSUE TITLE
Using win.on event in renderer process give error on console

### ISSUE THREAD

**Issue by AtiqGauri** [electron/electron#22545]
### Issue Details
Using events in renderer process throw error on development console
saying **Attempting to call a function in a renderer window that has been closed or released.**
```
let win = require('electron').remote.getCurrentWindow();
win.on('focus', ()=>{
    console.log('focused');
});
win.on('blur', ()=>{
    console.log('blured');
});
```

* **Electron Version:**
  * Electron 7.0.1 and 8.0.2
* **Operating System:**  
  * Windows 10

### Expected Behavior
It should run without error or warnings

### Actual Behavior
throw continuous warnings on console

### To Reproduce
https://gist.github.com/AtiqGauri/1cea1c548025faa77f9f29008ca5a5fe#file-main-js-L4

### Screenshots

![Screenshot (146)](https://user-images.githubusercontent.com/37737221/75960956-0efadf80-5ee7-11ea-933a-8f3c981ac05e.png)

**Comment by MarshallOfSound** [comment]
This is expected behavior.  You are registering a listener on a remote object, then reloading the window.  The context / page in which you registered the listener no longer exists but you did not clean up the listener.  This is Electron telling you that you haven't cleaned up correctly and a listener exists on a remote object from a renderer that does not exist any more.

TLDR: Avoid using `remote` and clean up your event handlers on `remote` objects if you **have** to use it
